common.h

```diff
@@ -148,6 +154,9 @@ typedef struct qos_frame qos_t;
 #define	QOS_SIZE (sizeof(qos_t))
 
 
+#if !defined __BYTE_ORDER || (__BIG_ENDIAN == __LITTLE_ENDIAN)
```

Comment:
(__BIG_ENDIAN == __LITTLE_ENDIAN)
Is this possible, eg. isn't this always false?

Reply:
I'm not shure about this. No chance to test it, but let's give it a try.

Reply:
The original problem was that neither of these macros were defined (with the underscores). Given this code, the pre-processor will substitute 0 for undefined variables, so it will be
```c
#if !defined __BYTE_ORDER || (0 == 0)
#error Shit hit the fan
#endif
```